test/EthValidatorsChecker.spec.ts

```diff
@@ -0,0 +1,375 @@
+            Buffer.from('', 'utf-8')
+          )
+      ).to.be.revertedWithCustomError(ethValidatorsChecker, 'InvalidVault')
+    })
+
+    it('fails for vault v1', async () => {
+      await expect(
+        ethValidatorsChecker
+          .connect(admin)
+          .checkValidatorsManagerSignature(
+            await vaultV1.getAddress(),
+            validatorsRegistryRoot,
+            Buffer.from('', 'utf-8'),
+            Buffer.from('', 'utf-8')
+          )
+      ).to.be.revertedWithCustomError(ethValidatorsChecker, 'InvalidVault')
+    })
+
+    it('fails for vault not collateralized not deposited', async () => {
+      await expect(
+        ethValidatorsChecker
+          .connect(admin)
+          .checkValidatorsManagerSignature(
+            await vaultNotDeposited.getAddress(),
+            validatorsRegistryRoot,
+            Buffer.from('', 'utf-8'),
+            Buffer.from('', 'utf-8')
+          )
+      ).to.be.revertedWithCustomError(ethValidatorsChecker, 'AccessDenied')
+    })
+
+    it('fails for signer who is not validators manager', async () => {
+      const { fakeValidatorsManager } = await validatorsManagerSetup()
+      const vaultAddress = await vault.getAddress()
+      const typedData = await getEthValidatorsCheckerSigningData(
+        keccak256(Buffer.concat(publicKeys)),
+        ethValidatorsChecker,
+        vault,
+        validatorsRegistryRoot
+      )
+      const signature = signTypedData({
+        privateKey: Buffer.from(ethers.getBytes(fakeValidatorsManager.privateKey)),
+        data: typedData,
+        version: SignTypedDataVersion.V4,
+      })
+      await expect(
+        ethValidatorsChecker
+          .connect(admin)
+          .checkValidatorsManagerSignature(
+            vaultAddress,
+            validatorsRegistryRoot,
+            Buffer.concat(publicKeys),
+            ethers.getBytes(signature)
+          )
+      ).to.be.revertedWithCustomError(ethValidatorsChecker, 'AccessDenied')
+    })
+
+    it('succeeds', async () => {
+      const { validatorsManager } = await validatorsManagerSetup()
+      const vaultAddress = await vault.getAddress()
+      const typedData = await getEthValidatorsCheckerSigningData(
+        Buffer.concat(publicKeys),
+        ethValidatorsChecker,
+        vault,
+        validatorsRegistryRoot
+      )
+      const signature = signTypedData({
+        privateKey: Buffer.from(ethers.getBytes(validatorsManager.privateKey)),
+        data: typedData,
+        version: SignTypedDataVersion.V4,
+      })
+
+      await expect(
+        ethValidatorsChecker
+          .connect(admin)
+          .checkValidatorsManagerSignature(
+            vaultAddress,
+            validatorsRegistryRoot,
+            Buffer.concat(publicKeys),
+            ethers.getBytes(signature)
+          )
+      ).to.eventually.be.greaterThan(0)
+    })
+  })
+
+  describe('check deposit data root', () => {
+    function getMultiProofArgs(options?: any) {
```

Comment:
better use `beforeEach`

Reply:
I need arguments here, It can not be simply replaced with beforeEach